app/src/main/java/com/gusta/wakemehome/DetailActivity.java

```diff
@@ -1,246 +1,356 @@
+
+    //temp png will be for unsaved snapshots on save it'll change to  map id.png
+    public static final String TEMP_IMAGE_FILE = "temp.png";
+    // map intent request code
+    private static final int MAP_REQUEST_CODE = 1;
+
+    //=========//
+    // MEMBERS //
+    //=========//
+
+    private AlarmEntry mAlarmEntry;                 // The current alarm entry
+    private DetailViewModel mViewModel;             // The current alarm view model
+    private ActivityDetailBinding mDetailBinding;   // The data binding object
+
+    //=========//
+    // METHODS //
+    //=========//
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mAlarmEntry = new AlarmEntry();
+        mAlarmEntry.setId(DEFAULT_ALARM_ID);
+        
+        mViewModel = ViewModelProviders.of(this).get(DetailViewModel.class);
+
+        // Init the data binding object
+        mDetailBinding = DataBindingUtil.setContentView(this, R.layout.activity_detail);
+
+        // Init the save button
+        Button mMapButton = mDetailBinding.OpenMapButton;
+        mMapButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                onOpenMapButtonClicked();
+            }
+        });
+
+        // Init the save button
+        Button mAlertButton = mDetailBinding.clockDetails.alert;
+        mAlertButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                onRingtoneSelect();
+            }
+        });
+
+        // Check for saved state (like after phone orientation change) - and load it
+        if (savedInstanceState != null) {
+            if(savedInstanceState.containsKey(INSTANCE_ALARM_ID))
+                mAlarmEntry.setId(savedInstanceState.getInt(INSTANCE_ALARM_ID, DEFAULT_ALARM_ID));
+            if(savedInstanceState.containsKey(INSTANCE_ALARM_MESSAGE))
+                mAlarmEntry.setMessage(savedInstanceState.getString(INSTANCE_ALARM_MESSAGE, ""));
+            if(savedInstanceState.containsKey(INSTANCE_ALARM_VIBRATE))
+                mAlarmEntry.setVibrate(savedInstanceState.getBoolean(INSTANCE_ALARM_VIBRATE, false));
+            if(savedInstanceState.containsKey(INSTANCE_ALARM_ALERT))
+                mAlarmEntry.setAlert(savedInstanceState.getString(INSTANCE_ALARM_ALERT, null));
+            if(savedInstanceState.containsKey(INSTANCE_ALARM_ADDRESS_DATA)){
+                MapAddress address = savedInstanceState.getParcelable(INSTANCE_ALARM_ADDRESS_DATA);
+                mAlarmEntry.setLocation(address.getLocation());
+                mAlarmEntry.setLatitude(address.getLatitude());
+                mAlarmEntry.setLongitude(address.getLongitude());
+                mAlarmEntry.setImage(address.getImage());
+                mAlarmEntry.setRadius(address.getRadius());

```

Comment:
This is obviously bad practice - will need to be updated for every new/updated/deleted field. In my version it persists without that.